src/staker.ts

```diff
@@ -0,0 +1,195 @@
+  /**
+   * The id of the NFT
+   */
+  tokenId: BigintIsh
+
+  /**
+   * Address to send rewards to.
+   */
+  recipient: string
+
+  /**
+   * The amount of `rewardToken` to claim. 0 claims all.
+   */
+  amount: BigintIsh
+}
+
+/**
+ * Options to specify when withdrawing a position.
+ */
+export interface WithdrawOptions {
+  /**
+   * Set when withdrawing. The position will be sent to `owner` on withdraw.
+   */
+  owner: string
+
+  /**
+   * Set when withdrawing. `data` is passed to `safeTransferFrom` when transferring the position from contract back to owner.
+   */
+  data: string
+}
+
+export abstract class Staker {
+  public static INTERFACE: Interface = new Interface(abi)
+
+  protected constructor() {}
+  private static INCENTIVE_KEY_ABI =
+    'tuple(address rewardToken, address pool, uint256 startTime, uint256 endTime, address refundee)'
+
+  /**
+   *  To claim rewards, must unstake and then claim.
+   * @param incentiveKey The unique identifier of a staking program.
+   * @param options Options for producing the calldata to claim. Can't claim unless you unstake.
+   * @returns The calldatas for 'unstakeToken' and 'claimReward'.
+   */
+  private static encodeClaim(incentiveKey: IncentiveKey, options: ClaimOptions): string[] {
+    const calldatas: string[] = []
+    calldatas.push(
+      Staker.INTERFACE.encodeFunctionData('unstakeToken', [
+        this._encodeIncentiveKey(incentiveKey),
+        toHex(options.tokenId)
+      ])
+    )
+    const recipient: string = validateAndParseAddress(options.recipient)
+    calldatas.push(
+      Staker.INTERFACE.encodeFunctionData('claimReward', [
+        incentiveKey.rewardToken.address,
+        recipient,
+        toHex(options.amount)
+      ])
+    )
+    return calldatas
+  }
+
+  /**
+   * Only claims rewards. Must unstake, claim, and then restake.
+   */
+  public static collectRewards(incentiveKey: IncentiveKey, options: ClaimOptions): MethodParameters {
+    const calldatas = this.encodeClaim(incentiveKey, options)
+    calldatas.push(
+      Staker.INTERFACE.encodeFunctionData('stakeToken', [
+        this._encodeIncentiveKey(incentiveKey),
+        toHex(options.tokenId)
+      ])
+    )
+    return {
+      calldata: Staker.INTERFACE.encodeFunctionData('multicall', [calldatas]),
+      value: toHex(0)
+    }
+  }
+
+  /**
+   *
+   * @param incentiveKeys A list of incentiveKeys to unstake from. Should include all incentiveKeys (unique staking programs) that `options.tokenId` is staked in.
+   * @param withdrawOptions Options for producing claim calldata and withdraw calldata. Can't withdraw without unstaking all programs for `tokenId`.
+   * @returns Calldata for unstaking, claiming, and withdrawing.
+   */
+  public static withdrawToken(
+    incentiveKeys: IncentiveKey | IncentiveKey[],
+    withdrawOptions: FullWithdrawOptions
+  ): MethodParameters {
+    let calldatas: string[] = []
+
+    incentiveKeys = incentiveKeys instanceof Array ? incentiveKeys : [incentiveKeys]
```

Comment:
per https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray#instanceof_vs_isarray let's use `Array.isArray` here and in `encodeDeposit`